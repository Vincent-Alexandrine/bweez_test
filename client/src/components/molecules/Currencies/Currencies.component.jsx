import React from 'react';
import PropTypes from 'prop-types';

import { Grid, Tile } from './Currencies.styled.js';

function Currencies({ items }) {
  return (
    <Grid>
      {items.map(({ name, market_cap, price, change, symbol }) => (
        <Tile key={symbol}>
          <h6 name="name">{name}</h6>
          <code name="symbol">{symbol}</code>
          <span name="price">{price}</span>
          <span name="change">{change}</span>
          <span name="cap">{market_cap}</span>
        </Tile>
      ))}
    </Grid>
  );
}

Currencies.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    market_cap: PropTypes.number,
    change: PropTypes.number,
    price: PropTypes.number,
    symbol: PropTypes.string
  })),
};

export default Currencies;
