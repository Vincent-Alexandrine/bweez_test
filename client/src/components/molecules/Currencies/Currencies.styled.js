import styled from 'styled-components';
import { getElevation } from 'Styles/methods/elevation';
import media from 'Styles/methods/media';

export const Grid = styled.ul`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(20rem, 1fr));
  grid-auto-rows: 1fr;
  grid-gap: 1rem;

  ${media.tablet`
    grid-template-columns: repeat(auto-fill, minmax(30%, 1fr));
  `}

  ${media.phone`
    grid-template-columns: repeat(auto-fill, minmax(45%, 1fr));
  `}

  list-style: none;
  margin: 0;
  padding: 0;

  &::before {
    content: '';
    width: 0;
    padding-bottom: 100%;
    grid-row: 1 / 1;
    grid-column: 1 / 1;
  }

  > *:first-child {
    grid-row: 1 / 1;
    grid-column: 1 / 1;
  }
`;

export const Tile = styled.li`
  padding: .5rem;
  border-radius: 4px;
  box-shadow: ${getElevation(2)};
  transition: all .1s ease-in;

  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  &:hover {
    box-shadow: ${getElevation(4)};
  }

  [name="name"], [name="price"] {
    flex: 0 1 60%;
  }
  [name="symbol"], [name="change"] {
    text-align: right;
    flex: 0 1 33%;
  }
  [name="price"], [name="change"] {
    align-self: center;
  }
  [name="cap"] {
    flex: 1 0 auto;
    align-self: flex-end;
  }
`;
