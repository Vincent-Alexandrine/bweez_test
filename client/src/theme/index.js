import ThemeService from './theme.service.js';

export default ThemeService;
