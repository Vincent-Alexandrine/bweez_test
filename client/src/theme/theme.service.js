/*
  getTheme(string) -> theme
  addTheme(string, theme) -> void || true || promise
  removeTheme(string) -> void || true || promise

  validateTheme(theme) -> boolean
*/

const themesMap = new Map();

class ThemeService {
  static instance;

  constructor() {
    if (ThemeService.instance)
      return ThemeService.instance;

    ThemeService.instance = this;
  }

  get(name) {
    return themesMap.get(name);
  }

  add({ name, ...theme }) {
    themesMap.set(name, theme);
  }

  names() {
    return [...themesMap.keys()];
  }

  remove(name) {
    return themesMap.delete(name);
  }
}

const singleton = new ThemeService();

export default singleton;
