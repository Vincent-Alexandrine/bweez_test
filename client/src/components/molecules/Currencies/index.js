import Currencies from './Currencies.component.jsx';

export default Currencies;
