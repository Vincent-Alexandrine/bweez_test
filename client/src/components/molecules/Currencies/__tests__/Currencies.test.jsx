import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import theme from 'Styles/vManuscript';

import Currencies from '../Currencies.component.jsx';
import { Grid, Tile } from '../Currencies.styled.js';

test('Currencies component should match snapshot', () => {
  const component = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Currencies items={[]} />
      </ThemeProvider>
    );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('Currencies styled component should match snapshot', () => {
  const component = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Grid>
          <Tile />
        </Grid>
      </ThemeProvider>
    );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
