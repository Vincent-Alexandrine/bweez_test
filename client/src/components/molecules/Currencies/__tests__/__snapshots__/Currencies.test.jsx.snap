// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currencies component should match snapshot 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(20rem,1fr));
  grid-auto-rows: 1fr;
  grid-gap: 1rem;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c0::before {
  content: '';
  width: 0;
  padding-bottom: 100%;
  grid-row: 1 / 1;
  grid-column: 1 / 1;
}

.c0 > *:first-child {
  grid-row: 1 / 1;
  grid-column: 1 / 1;
}

@media (max-width:768px) {
  .c0 {
    grid-template-columns: repeat(auto-fill,minmax(30%,1fr));
  }
}

@media (max-width:576px) {
  .c0 {
    grid-template-columns: repeat(auto-fill,minmax(45%,1fr));
  }
}

<ul
  className="c0"
/>
`;

exports[`Currencies styled component should match snapshot 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(20rem,1fr));
  grid-auto-rows: 1fr;
  grid-gap: 1rem;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c0::before {
  content: '';
  width: 0;
  padding-bottom: 100%;
  grid-row: 1 / 1;
  grid-column: 1 / 1;
}

.c0 > *:first-child {
  grid-row: 1 / 1;
  grid-column: 1 / 1;
}

.c1 {
  padding: .5rem;
  border-radius: 4px;
  box-shadow: 0 4.666666666666667px 6px -1px rgba(0,0,0,0.1),0 2px 2px 0px rgba(0,0,0,0.07),0 1px 4.5034985777834295px 0px rgba(0,0,0,0.06);
  -webkit-transition: all .1s ease-in;
  transition: all .1s ease-in;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1:hover {
  box-shadow: 0 3.333333333333333px 12px -2px rgba(0,0,0,0.1),0 4px 5px 0px rgba(0,0,0,0.07),0 2px 7.789324958007403px 1px rgba(0,0,0,0.06);
}

.c1 [name="name"],
.c1 [name="price"] {
  -webkit-flex: 0 1 60%;
  -ms-flex: 0 1 60%;
  flex: 0 1 60%;
}

.c1 [name="symbol"],
.c1 [name="change"] {
  text-align: right;
  -webkit-flex: 0 1 33%;
  -ms-flex: 0 1 33%;
  flex: 0 1 33%;
}

.c1 [name="price"],
.c1 [name="change"] {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 [name="cap"] {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

@media (max-width:768px) {
  .c0 {
    grid-template-columns: repeat(auto-fill,minmax(30%,1fr));
  }
}

@media (max-width:576px) {
  .c0 {
    grid-template-columns: repeat(auto-fill,minmax(45%,1fr));
  }
}

<ul
  className="c0"
>
  <li
    className="c1"
  />
</ul>
`;
